### ISSUE TITLE
400 PUT Error *AFTER V2 Upgrade*

### ISSUE THREAD

**Issue by tedtms** [laurent22/joplin#5123]
When synchronizing from Joplin Desktop to Joplin Server, synchronization hangs. When testing the synchronization settings, a 400 PUT denial error is received. Started happening after upgrade from Joplin v1.x to Joplin v2.1.1

## Environment
**Desktop:**
Joplin 2.0.11 (prod, win32)
Sync Version: 2
Profile Version: 39
Keychain Supported: Yes
Revision: 981b1d845

**Server:**
Joplin 2.0.11
Using joplin/server:latest docker image

**Platform:**
OS: Ubuntu 20.04

## Steps to reproduce

1. Perform upgrade preparations [as per V2 upgrade instructions](https://discourse.joplinapp.org/t/major-breaking-change-in-coming-joplin-server-2-0/17254)
2. Upgrade Docker to v2 image (I'm using server:latest)
3. Upgrade desktop to v2
4. Attempt re-sync of all items from desktop client to server
5. Result: `Not allowed: PUT (Code 400)`

## Describe what you expected to happen
Items should re-sync with upgraded server successfully.

## Logfile
[debug.txt](https://github.com/laurent22/joplin/files/6720836/debug.txt)

**Comment by laurent22** [comment]
If you are using Joplin Server beta it's best to use the latest pre-releases too: https://github.com/laurent22/joplin/releases

I don't know what error you have without looking at a complete log, but it might just be an incompatibility that has been resolved in one of the latest pre-releases.  Things are a bit unstable at the moment as Joplin Server is still quite new but it should gradually get better over time. Already the next version includes a check on the api version, to allow displaying a more useful message in case of incompatibility.

**Comment by ameliorate-droid** [comment]
Using iOS App v 12.0.2 with the Joplin Server using docker image from joplin/server:latest.

When testing the synchronization settings, a 400 PUT denial error is received. Started happening after upgrade of iOS App.
The desktop version 1.7.11(prod, linux) doesn't have this issue.

Docker logs and iOS screenshot attached.

[joplinSyncErrorLog.txt](https://github.com/laurent22/joplin/files/6721485/joplinSyncErrorLog.txt)
![joplinSyncErrorOnIOS](https://user-images.githubusercontent.com/52375566/123542702-70a32500-d78e-11eb-9b69-7806f3ceda1d.jpg)

**Comment by ameliorate-droid** [comment]
On the iOS app, the sync option with Joplin Server is with the beta only. I can't remember seeing the beta in this list earlier.

![joplinIOSSyncOptions](https://user-images.githubusercontent.com/52375566/123543542-6420cb80-d792-11eb-9b40-a5d790d4d907.png)

**Comment by aliakseiz** [comment]
```
joplin-server | 2021-06-27 19:06:49: App: GET /api/items/root:/locks/*:/children (3ms)
joplin-server | 2021-06-27 19:06:49: [error] App: 400: PUT /api/batch_items : Not allowed: PUT 
joplin-server | 2021-06-27 19:06:49: App: PUT /api/batch_items (2ms)
```

Had same error message with `latest` server and `2.1.7` client.
Issue resolved after upgrading the server to `2.1.6-beta`.

**Comment by tedtms** [comment]
> ```
> joplin-server | 2021-06-27 19:06:49: App: GET /api/items/root:/locks/*:/children (3ms)
> joplin-server | 2021-06-27 19:06:49: [error] App: 400: PUT /api/batch_items : Not allowed: PUT 
> joplin-server | 2021-06-27 19:06:49: App: PUT /api/batch_items (2ms)
> ```
> 
> Had same error message with `latest` server and `2.1.7` client.
> Issue resolved after upgrading the server to `2.1.6-beta`.

That did it for me, thanks a ton!
